addons/mrp/models/stock_move.py

```diff
@@ -342,12 +342,14 @@ def _key_assign_picking(self):
     def _prepare_merge_moves_distinct_fields(self):
         distinct_fields = super()._prepare_merge_moves_distinct_fields()
         distinct_fields.append('created_production_id')
+        distinct_fields.append('bom_line_id')
```

Comment:
The issue seems to be here
In master we have a new way to choose candidate moves for merge
https://github.com/odoo/odoo/blob/9ea04fa4b2024e1d4cf2e11c64e6c261ed84d8d6/addons/mrp/models/stock_move.py#L413
which is including all production related moves, thing that did not occur in 14

To workaround the issue we may avoid this part of the fix, although not critical this is here to avoid possible issue when mixing kit product with components in a pos order, but after a quick test it seems everything is working as expected

Reply:
Removing this and the other line reference to `bom_line_id` works. Looks like the `_update_candidate_moves_list` changed the behaviour of `_merge_moves` for moves generated from `mrp`. Perhaps that's the right fix instead of setting distinct fields. (I don't know.) But I agree that for this fix to proceed, we can just remove the two lines using `bom_line_id`.